frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionList.jsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { t } from "ttag";
+
+import EmptyState from "metabase/components/EmptyState";
+import Search from "metabase/entities/search";
+
+import { QuestionPickerItem } from "./QuestionPickerItem";
+
+import { EmptyStateContainer } from "./QuestionList.styled";
+
+QuestionList.propTypes = {
+  searchText: PropTypes.string,
+  collectionId: PropTypes.number,
+  onSelect: PropTypes.func.isRequired,
+  hasCollections: PropTypes.bool,
+};
+
+export function QuestionList({
+  searchText,
+  collectionId,
+  onSelect,
+  hasCollections,
+}) {
+  if (collectionId === "personal" && !searchText) {
+    return null;
+  }
+
+  const trimmedSearchText = searchText.trim();
+  const isSearching = !!trimmedSearchText;
+
+  let query = isSearching
+    ? { q: trimmedSearchText }
+    : { collection: collectionId };
+
+  query = {
+    ...query,
+    models: "card",
+  };
+
+  return (
+    <Search.ListLoader entityQuery={query} wrapped>
+      {({ list }) => {
+        if (list.length === 0 && (isSearching || !hasCollections)) {
```

Comment:
Does this translate a situation that can be immediately explained by moving it to a const?

Roughly like 

`const thisIsSituationXInTheInterface = list.length === 0 && (isSearching || !hasCollections`